src/analytics.ts

```diff
@@ -7,10 +7,18 @@ export type Geo = {
   region?: string;
   ip?: string;
 };
+
+/**
+ * denotes the success field in the analytics submission.
+ * Set to true when ratelimit check passes. False when request is ratelimited.
+ * Set to "denied" when some request value is in deny list.
+ */
+export type EventSuccess = boolean | "denied"
```

Comment:
You know what `"pass" | "ratelimited" | "denied"` would be fantastic IMO. Definitely more expressive.

Reply:
this would also fix the [`1 | null | denied` in core analytics](https://github.com/upstash/core-analytics/pull/37#discussion_r1620341827), but it would be backwards incompatible, users wouldn't be able to view past analytics at all after we update the dashboard.

I think we can do it if you are okay with backwards compatibility

Reply:
Agreed. Let's not forget to create a ticket for this.

Reply:
[ticket](https://linear.app/upstash/issue/DX-966/ratelimit-success-conditions-update)